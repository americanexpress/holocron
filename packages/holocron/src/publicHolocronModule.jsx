import { getModuleName } from './holocronModule';

// TODO remove this public holocronModule in next major version
export default function holocronModule(holocronConfig = {}) {
  return function addHolocronConfig(WrappedComponent) {
    console.warn(`'holocronModule' has been deprecated in favor of the Holocron Config API. Please migrate to Holocron Config API in ${getModuleName(WrappedComponent)}.`);
    // Need to pass through to the new Holocron Config API
    // eslint-disable-next-line no-param-reassign
    WrappedComponent.holocron = holocronConfig;
    return WrappedComponent;
  };
}
