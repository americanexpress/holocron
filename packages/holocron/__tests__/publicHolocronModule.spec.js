import publicHolocronModule from '../src/publicHolocronModule';

const warn = jest.spyOn(console, 'warn');

const holocronProperties = {
  name: 'MyModule',
  reducer: function fakeReducer() {},
  load: function fakeLoad() {},
  shouldModuleReload: function fakeShouldModuleReload() {},
  loadModuleData: function fakeLoadModuleData() {},
  mergeProps: function fakeMergeProps() {},
  options: function fakeOptions() {},
};

describe('publicHolocronModule', () => {
  it('should add holocron config properties to component', () => {
    const FakeComponent = {
      name: 'FakeComponent',
      displayName: 'FakeComponent',
    };
    expect(publicHolocronModule({
      ...holocronProperties,
    })(FakeComponent)).toMatchSnapshot();
  });

  it('should attach empty holocron config to component without args', () => {
    const FakeComponent = {
      name: 'FakeComponent',
      displayName: 'FakeComponent',
    };
    expect(publicHolocronModule()(FakeComponent)).toMatchSnapshot();
  });

  it('should send a deprecation warning', () => {
    const FakeComponent = {
      name: 'FakeComponent',
      displayName: 'FakeComponent',
    };
    publicHolocronModule()(FakeComponent);
    expect(warn).toHaveBeenCalled();
  });
});
