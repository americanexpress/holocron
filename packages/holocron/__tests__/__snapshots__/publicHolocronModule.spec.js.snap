// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`publicHolocronModule should add holocron config properties to component 1`] = `
Object {
  "displayName": "FakeComponent",
  "holocron": Object {
    "load": [Function],
    "loadModuleData": [Function],
    "mergeProps": [Function],
    "name": "MyModule",
    "options": [Function],
    "reducer": [Function],
    "shouldModuleReload": [Function],
  },
  "name": "FakeComponent",
}
`;

exports[`publicHolocronModule should attach empty holocron config to component without args 1`] = `
Object {
  "displayName": "FakeComponent",
  "holocron": Object {},
  "name": "FakeComponent",
}
`;
